### PR TITLE
Update ModelMesh version to `v0.10.0`

### PR DESCRIPTION
**Pull request by ckadner**
#### Proposed Changes

Update to the latest ModelMesh version (`v0.10.0`)

> If your changes should also be in the most recent release, add the corresponding "cherrypick-0.X"
label to the original PR; for example, "cherrypick-0.10".

#### TODOs:

> - [ ] Best practice is to open a PR for the cherry-pick yourself after your original PR has been merged
into the main branch.
> - [ ] After the cherry-pick PR has merged, remove the cherry-pick label from the original PR.

**Type of changes**

- [x] This change requires a documentation update --> https://github.com/kserve/website/pull/214

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

YES

**Checklist**:

- [x] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
???
```
Not sure, maybe, 
- "Updgrade to ModelMesh v0.10.0" 
or
- point to ModelMesh release notes here
https://github.com/kserve/modelmesh-serving/releases/tag/v0.10.0

---

/cc @yuzisun @rachitchauhan43 @njhill

### CODE REVIEW

**Comment by yuzisun** [comment]
@ckadner Did other modelmesh artifacts get changed? like serving runtimes or modelmesh controller yamls

**Comment by ckadner** [comment]
> @ckadner Did other modelmesh artifacts get changed? like serving runtimes or modelmesh controller yamls

@yuzisun -- there have been a few changes since the ModelMesh `v0.9.0` release that affect our YAML files:

https://github.com/kserve/modelmesh-serving/commits/main/config?since=2022-07-20

Some more notable/impactful ones, I suppose, are:
- [feat: TorchServe support #250](https://github.com/kserve/modelmesh-serving/pull/250)
- [fix: Fix namespace-scoped install #212](https://github.com/kserve/modelmesh-serving/pull/212/files)
- [feat: Make list of supported built-in runtime server types overridable #220](https://github.com/kserve/modelmesh-serving/pull/220)
- [chore: Update to use latest OVMS 2022.2 image #258](https://github.com/kserve/modelmesh-serving/pull/258/files)
- [feat: Update install to support ClusterServingRuntime #252](https://github.com/kserve/modelmesh-serving/pull/252)
- [feat: Enable HA by default for Modelmesh Controller #272](https://github.com/kserve/modelmesh-serving/pull/272/files)
- [feat: Support passing labels and annotations from ServingRuntimePodSpec to runtime Pod](https://github.com/kserve/modelmesh-serving/pull/271)
- [feat: Supports image pull secrets #276](https://github.com/kserve/modelmesh-serving/pull/276/files)

Are there some files that get copied/synced/overlayed between KServe and ModelMesh that I should look at and do a file based compare?

@njhill

**Comment by ckadner** [comment]
@njhill @yuzisun -- I think I got the helm charts updated for all of the `config` changes that went into ModelMesh `v0.10.0`

For reference, these are the file changes in ModelMesh (`config`) between `v0.9.0` and `v0.10.0`:
https://github.com/kserve/modelmesh-serving/compare/v0.9.0..v0.10.0#diff-8a88de8f93e37606e339d622f64e288408f314acaa6e9abb3895632428252059

**Comment by njhill** [comment]
Thanks @ckadner, these changes look great to me!

File: charts/kserve-resources/templates/servingruntimes.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: serving.kserve.io/v1alpha1
-kind: ServingRuntime
+kind: ClusterServingRuntime
```
Comment (yuzisun) [inline charts/kserve-resources/templates/servingruntimes.yaml]: What happens when this is changed from `ServingRuntime` to `ClusterServingRuntime` during upgrade?

Comment (ckadner) [reply charts/kserve-resources/templates/servingruntimes.yaml]: that's a good question. For @njhill ? :-)

I realized they should live in the `clusterservingruntimes.yaml` file now, no longer in `servingruntimes.yaml`. 

I just updated the [failing helm](https://github.com/kserve/kserve/actions/runs/3916080535/jobs/6695334449#step:6:189) test accordingly:

```
Error from server (NotFound): servingruntimes.serving.kserve.io "mlserver-0.x" not found
```

**Comment by ckadner** [comment]
`test-with-helm` failed again. This time with this [error](https://github.com/kserve/kserve/actions/runs/3916512430/jobs/6696059417#step:7:1622):

```
helm/test_model_mesh_sklearn.py:74:
> res = predict_modelmesh(service_name, "./data/mm_sklearn_input.json", pod_name)`
...
=========================== short test summary info ============================
FAILED helm/test_model_mesh_sklearn.py::test_sklearn_modelmesh - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
====== 1 failed, 1 passed, 56 deselected, 3 warnings in 421.67s (0:07:01) ======
Error: Process completed with exit code 1.
```

```Python
def test_sklearn_modelmesh():
    ...
    kserve_client.wait_isvc_ready(service_name)
    pods = kserve_client.core_api.list_namespaced_pod("default", label_selector="name=modelmesh-serving-mlserver-0.x")

    pod_name = pods.items[0].metadata.name
    res = predict_modelmesh(service_name, "./data/mm_sklearn_input.json", pod_name)
    ...
```

```Python
def predict_modelmesh(service_name, input_json, pod_name, model_name=None):
    with open(input_json) as json_file:
        data = json.load(json_file)

        if model_name is None:
            model_name = service_name
        with portforward.forward("default", pod_name, 8008, 8008):
            url = f"http://localhost:8008/v2/models/{model_name}/infer"
            response = requests.post(url, json.dumps(data))
            return json.loads(response.content.decode("utf-8"))
```

I assume the `ServingRuntime` vs `ClusterServingRuntime` change is the issue here as well. Not sure if this line will still work:
```
pods = kserve_client.core_api.list_namespaced_pod(...)
```

**Review by njhill — APPROVED** [review]
LGTM thanks @ckadner!

**Comment by yuzisun** [comment]
Thanks @ckadner !!
/lgtm
/approve

File: charts/kserve-resources/templates/servingruntimes.yaml (hunk 1 above)
Comment (ckadner) [reply charts/kserve-resources/templates/servingruntimes.yaml]: @rafvasq did a test on the migration scenario yesterday. Paraphrasing a bit:

> deployed 0.9 and an `isvc`, then upgraded to `0.10`. The `isvc` is untouched (relevant pods redeployed as expected), while both `clusterservingruntimes` and `servingruntimes` are now available. The `servingruntimes` from `v0.9` weren't deleted. The `servingruntimes` still exist (along with the `clusterservingruntimes`)
> then did manually delete the `servingruntimes` to see what happened --> no changes to any pods or `isvc`s


Thanks Rafael!

**Comment by ckadner** [comment]
Thanks @yuzisun  (and @njhill  for the review)!

**Pull request merged**